### PR TITLE
[Android] Add Status handling for onReport code path

### PR DESCRIPTION
**Pull request by yunhanw-google**
-- Add the map for attribute status and the map for a array of event status in java cluster state 
-- Whenever there is status coming through report path in JNI, we would update the corresponding in java cluster state.
-- We cannot pass the status via onError path becuase onError path is handling the fatal error per contract.
-- Add the corresponding tests.

### CODE REVIEW

**Review by joonhaengHeo — APPROVED** [review]

File: examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImReadCommand.kt [hunk 1]
```diff
@@ -93,6 +96,11 @@ class PairOnNetworkLongImReadCommand(
           "No local config disabled attribute found."
         }
 
+      val unitTestCluster =
+        requireNotNull(endpointOne.getClusterState(UNIT_TEST_CLUSTER)) {
+          "unit test cluster not found."
```
Comment (yufengwangca) [inline examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImReadCommand.kt:101]: capitalize 'Unit'

File: src/controller/java/src/chip/devicecontroller/model/Status.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  private Integer status;
+  private Optional<Integer> clusterStatus;
+
+  private Status(int status, Optional<Integer> clusterStatus) {
+    this.status = status;
+    this.clusterStatus = clusterStatus;
+  }
+
+  // Getters
+  public int getStatus() {
```
Comment (yufengwangca) [inline src/controller/java/src/chip/devicecontroller/model/Status.java:34]: The method public int getStatus() returns a primitive int type. This decision ensures that the method will always return a non-null integer value. When calling getStatus(), the caller is guaranteed to receive an int value without needing to handle the possibility of null. But status is defined as Integer

Comment (yunhanw-google) [reply src/controller/java/src/chip/devicecontroller/model/Status.java]: updated, thanks

File: examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImReadCommand.kt (hunk 1 above)
Comment (yunhanw-google) [reply examples/java-matter-controller/java/src/com/matter/controller/commands/pairing/PairOnNetworkLongImReadCommand.kt]: done, thanks

**Review by chulspro — APPROVED** [review]
Looks good to me

**Pull request merged**